src/drawing/DrawingAnnotationsContainer.tsx

```diff
@@ -27,24 +29,30 @@ export type Props = {
     annotations: AnnotationDrawing[];
     drawnPathGroups: Array<PathGroup>;
     isCreating: boolean;
+    status: CreatorStatus;
 };
 
 export const mapStateToProps = (state: AppState, { location }: { location: number }): Props => {
+    const creatorStatus = getCreatorStatus(state);
+
     return {
         activeAnnotationId: getActiveAnnotationId(state),
         annotations: getAnnotationsForLocation(state, location).filter(isDrawing),
         drawnPathGroups: getDrawingDrawnPathGroupsForLocation(state, location),
         isCreating:
             isFeatureEnabled(state, 'drawingCreate') &&
             getAnnotationMode(state) === Mode.DRAWING &&
-            getCreatorStatus(state) !== CreatorStatus.pending,
+            creatorStatus !== CreatorStatus.pending,
```

Comment:
It's usually a bit of a code smell when we both derive a value from state and also pass the base value into the component. It seems like `isCreating` is also getting quite overloaded at this point.